.github/workflows/ci.yml

```diff
@@ -99,3 +101,30 @@ jobs:
       - name: pytest
         run: python3 -m pytest -x -v
 
+      # Note in order for ctest -T Coverage to work the tests have to be executed first
+      # with ctest. Moving this before the test step breaks coverage
+      - name: coverage
+        if: contains(matrix.compiler, 'g++-10')
+        run: |
+          ctest -T test -T Coverage
+          gcovr -r ../ . --html -o coverage-report.html
+        working-directory: ./build
+
+      - name: archive coverage results
+        if: contains(matrix.compiler, 'g++-10')
+        uses: actions/upload-artifact@v2
+        with:
+          name: code-coverage-report
+          path: ./build/coverage-report.html
+
+
+      - name: sanitizers
+        if: contains(matrix.compiler, 'clang++-12')
+        run: |
+          cmake -S .. -B . -DCMAKE_BUILD_TYPE=Debug -DENABLE_ADDRESS_SANITIZER=ON -DENABLE_SANITIZER_ADDRESS=ON -DENABLE_SANITIZER_UNDEFINED_BEHAVIOR=ON
```

Comment:
I think we should avoid in-source builds, it may mess things up in the future.

Reply:
This is not an in source build. I changed it to make it clearer.